schedule/scheduler.go

```diff
@@ -205,6 +206,7 @@ func (s *Scheduler) StartJob(id JobID) error {
 	// clear any error messages if we are starting the job
 	job.errorMsg = ""
 	if job.running {
+		job.lk.Unlock()
```

Comment:
Could just defer after line 205. Not much else happens in this function so it won't be locked for long